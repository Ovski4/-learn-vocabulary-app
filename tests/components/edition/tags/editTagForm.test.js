import reducer from '../../../../src/reducers/index';
import { createStore } from 'redux';
import React from 'react';
import EditTagForm from '../../../../src/components/edition/tags/editTagForm/connect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

describe('EditTagForm component', () => {

    it('Should render the component', () => {

        const store = createStore(reducer, {
            tags: [
                {
                    id: '1',
                    label: 'test'
                }
            ],
            config: {
                locale: 'en'
            }
        });

        const rendered = renderer.create(
            <Provider store={store}>
                <EditTagForm tag={{id: '1', label: 'test'}} />
            </Provider>
        ).toJSON();
        expect(rendered).toMatchSnapshot();
    });

});
