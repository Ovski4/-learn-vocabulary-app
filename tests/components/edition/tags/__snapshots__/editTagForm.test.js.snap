// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTagForm component Should render the component 1`] = `
<View
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderTopWidth": 0.4,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "margin": 5,
          "width": "15%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Label:
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#eeeeee",
          "borderRadius": 2,
          "flex": 1,
          "margin": 5,
          "paddingLeft": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue="test"
        onChangeText={[Function]}
        style={
          Object {
            "fontSize": 16,
            "height": 32,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": "15%",
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 5,
          "marginTop": 5,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
